FrameworkSystem/Client/ComponentInstaller.py

```diff
@@ -195,40 +196,41 @@ def loadDiracCfg( self, verbose = False ):
     self.mysqlRootPwd = self.localCfg.getOption( cfgInstallPath( 'Database', 'RootPwd' ), self.mysqlRootPwd )
     if verbose and self.mysqlRootPwd:
       gLogger.notice( 'Reading Root MySQL Password from local configuration' )
+    else:
+      gLogger.warn( 'MySQL root password not found' )
 
-    self.mysqlUser = self.localCfg.getOption( cfgInstallPath( 'Database', 'User' ), '' )
-    if self.mysqlUser:
-      if verbose:
-        gLogger.notice( 'Reading MySQL User from local configuration' )
+    self.mysqlUser = self.localCfg.getOption( cfgInstallPath( 'Database', 'User' ), self.mysqlUser )
+    if verbose and self.mysqlUser:
```

Comment:
IIUC it was done so that depending on the caller, different levels could be set. But I agree it's a bit weird way of working. I don't want to change this code right now as it's quite a pain to test it, so not for this release.